### PR TITLE
Add: newline character when saving files

### PR DESCRIPTION
**Pull request by rbeer**
Hi there!

First of all, thank you for this great module! Love it.

With .save() created .json files, though, are missing a newline character at the end. Considering that config files are often processed automatically, one might use UNIX tools to do so. According to the POSIX standard for a line, those tools often expect that newline, even worse sometimes fail when it's missing.

http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Have a great day! :)

### CODE REVIEW

**Comment by mhamann** [comment]
@rbeer would you be interested in adding some tests for this so we can merge it?

**Comment by rbeer** [comment]
Can't give you an estimate when I get around to do it, but I'll have it on my list.

**Comment by Alexsey** [comment]
@rbeer did you really run in to the issue that you are using some UNIX program that fails due to the non-new-line file ending? Or it is just breaks your code style? If it is a question of a code style than you should add an option that would determine should the EOL be inserted or not. Otherwise making it mandatory is actually the same as not having it at all

**Comment by mhamann** [comment]
@Alexsey here's some good explanation: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

More or less, I see the addition of a newline char as a good thing, since that tends to be a pretty standardized Unix/Linux convention.

Are there cases you are thinking of where this would be undesirable?

**Comment by Alexsey** [comment]
@mhamann The answer you are pointing out soon will be 10 years old. In JS world you will not encouraged so old UNIX utils that would have problems with no new line at the end of file. I'm pretty sure that as for the end of 2017 for JavaScript it is just a question of style and not more. If I'm wrong - please point out any of your JS arsenal that will suffer from it

**Comment by rbeer** [comment]
Well, let me put it this way: I just tried to create a file with no EOL at the end - to no avail. :joy:
Emacs, nano, vim - they all force an EOL char onto the last line.

Anyway. The main point is that it is the POSIX standard. Just because Windows (maybe) doesn't care, doesn't mean that we shouldn't either.
@Alexsey Please read the box in the first answer over at the linked SO post. The issue isn't that the post is 10 years old, but rather that the standard is still in effect.
Also, JSON might stand for JavaScript Object Notation, but it "matured" to become a generally used data interchange format. So no; JSON isn't used only by JavaScript applications, but many other more.

In conclusion: I'm not arguing for my personal style flavor here. It's a backwards argument to say that there might be programs out there that are confused by EOL at EOF. They would be ones doing it wrong. I'd rather let them know, than doing it wrong too, just to be compatible with them.

**Comment by Alexsey** [comment]
@rbeer Emacs, and vim may be setup any way you like. Pretty sour that nano as well. the POSIX standard is that there is a [3.206 Line definition](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206) that tells what the line is. It has nothing about EOF - it is about EOL. But also there is a [3.195 Incomplete Line definition](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_195). That defines a line at the EOF that has no termination symbol at the EOL. They are all just a definitions. As long that there are no issues you should not wary about it. And as I say - I'm not against using line terminator at the EOF - I'm only about making this thing configurable because there are different styles. And different styles is really on of the actual problems of today's JS world

**Comment by rbeer** [comment]
Remade in #294, closing.
